### DISCUSSION TITLE
FastAPI > 0.95 raises AssertionError: Cannot use `Query` for path param

### DISCUSSION THREAD

**Discussion by kujiy** [fastapi/fastapi#9507] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Query

@router.get("/users/{userId}")
async def users_api(
    userId: Optional[str] = Query("test999", min_length=5, regex=r'[a-z]+\d+', example="abc999")
):
    return userId

error:
E   AssertionError: Cannot use `Query` for path param 'userId'
```

### Description

FastAPI <= 0.94.1 supported `fastapi.Query` for a path param but FastAPI <= 0.95 does not. What's the alternative way?

### Operating System

Linux, macOS

### Operating System Details

docker hub python official images
Mac OS 10.15

### FastAPI Version

0.95.1

### Python Version

Python 3.10.4

### Additional Context

[The doc](https://fastapi.tiangolo.com/tutorial/path-params/) indicates validating a path param inside the controller function. However, it was helpful to validate it with a Query object. Is there a good way to achieve the same validation against a path param?

**Comment by chrisK824** [comment]
Hi @kujiy , have you tried to convert `Query` to `Path` parameter?
This changed also broke some of mine endpoints as well, removing the Query declaration made it back working, as recognising it as Path parameter

https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/#number-validations-greater-than-or-equal

**Reply by chrisK824** [reply]
@kujiy have you tried that?

**Reply by kujiy** [reply]
sorry for my late. An error occurs when adding a default value.

```
    assert default is ..., "Path parameters cannot have a default value"
E   AssertionError: Path parameters cannot have a default value
```

But we must not use a default value, so it looks fine. Let me check more.

**Reply by chrisK824** [reply, marked as answer]
Cool @kujiy , I think default values are not allowed anyway there. If that covers your question would appreciate if you mark that as an answer, cheers!
